content/develop/interact/search-and-query/advanced-concepts/vectors.md

```diff
@@ -9,369 +9,503 @@ categories:
+redis_client.hset('docs:01', mapping = {"vector": vector_bytes, "foo": "bar"})
 ```
-FT.CREATE my_index ON JSON SCHEMA $.vec as vector VECTOR FLAT 6 TYPE FLOAT32 DIM 4 DISTANCE_METRIC L2 
-```
+<br/>
+{{% alert title="Tip" color="warning" %}}
+Note that the vector blob size must match the vector field dimension and float type specified in the schema, otherwise the indexing will fail in the background.
+{{% /alert  %}}
+<br/>
+
+### JSON
+Store or update vectors and any metadata in [JSON]({{< baseurl >}}/develop/data-types/json/) with the [`JSON.SET`]({{< baseurl >}}/commands/json.set/) command.
 
-Unlike in hashes, vectors are stored in JSON documents as arrays (not as blobs).
+Unlike in hashes, vectors are stored in JSON documents as arrays (not as bytes).
 
 **Example**
 ```
-JSON.SET 1 $ '{"vec":[1,2,3,4]}'
+JSON.SET docs:01 $ '{"vector":[0.34,0.63,-0.54,-0.69,0.98,0.61], "foo": "bar"}'
 ```
 
-As of v2.6.1, JSON supports multi-value indexing. This capability accounts for vectors as well. Thus, it is possible to index multiple vectors under the same JSONPath. Additional information is available in the [Indexing JSON documents]({{< baseurl >}}/develop/interact/search-and-query/indexing/#index-json-arrays-as-vector) section. 
+**One of the additional benefits of JSON is schema flexibility.** As of v2.6.1, JSON supports multi-value indexing.
+Thus, it is possible to index multiple vectors under the same JSONPath.
+
+Here are a some examples of multi-value indexing with vectors:
 
 **Example**
 ```
-JSON.SET 1 $ '{"vec":[[1,2,3,4], [5,6,7,8]]}'
-JSON.SET 1 $ '{"foo":{"vec":[1,2,3,4]}, "bar":{"vec":[5,6,7,8]}}'
+JSON.SET key $ '{"vector":[[1,2,3,4], [5,6,7,8]]}'
+JSON.SET key $ '{"item1":{"vector":[1,2,3,4]}, "item2":{"vector":[5,6,7,8]}}'
 ```
 
-## Querying vector fields
+Additional information and examples are available in the [Indexing JSON documents]({{< baseurl >}}/develop/interact/search-and-query/indexing/#index-json-arrays-as-vector) section.
 
-You can use vector similarity queries in the [`FT.SEARCH`]({{< baseurl >}}/commands/ft.search/) query command. To use a vector similarity query, you must specify the option `DIALECT 2` or greater in the command itself, or set the `DEFAULT_DIALECT` option to `2` or greater, by either using the command [`FT.CONFIG SET`]({{< baseurl >}}/commands/ft.config-set/) or when loading the `redisearch` module and passing it the argument `DEFAULT_DIALECT 2`.
+## Search with vectors
+You can run vector search queries with the [`FT.SEARCH`]({{< baseurl >}}/commands/ft.search/) or [`FT.AGGREGATE`]({{< baseurl >}}/commands/ft.aggregate/) commands.
 
-There are two types of vector queries: *KNN* and *range*:
+To use a vector similarity query with `FT.SEARCH`, you must always:
+- Specify the option `DIALECT 2` or greater in the search command at runtime.
+- Or, set the `DEFAULT_DIALECT` option to `2` or greater by modifying the search configuration with [`FT.CONFIG SET`]({{< baseurl >}}/commands/ft.config-set/).
```

Comment:
I think this can be made slightly more prominant with an explination of why dialects exist. I had a customer looking at one of our docs sets and was super confused by the dialect things so that's where this ask is coming from

Reply:
Yeah, we should link out to the general section on dialects, because it's definitely a foot gun